modules/rostests/apitests/gdi32/StretchBlt.c

```diff
@@ -18,91 +18,852 @@
+    ok(!memcmp(dstBuffer, expected, 16),
+       "StretchBlt expected { %08X, %08X, %08X, %08X } got { %08X, %08X, %08X, %08X }\n",
+        expected[0], expected[1], expected[2], expected[3],
+        dstBuffer[0], dstBuffer[1], dstBuffer[2], dstBuffer[3] );
+
+    SelectObject(hdcSrc, oldSrc);
+    DeleteObject(bmpSrc);
+
+    biSrc.bmiHeader.biBitCount = 1;
+    bmpSrc = CreateDIBSection(hdcScreen, &biSrc, DIB_RGB_COLORS, (void**)&srcBuffer, NULL, 0);
+    oldSrc = SelectObject(hdcSrc, bmpSrc);
+    *((DWORD *)colors + 0) = 0x123456;
+    *((DWORD *)colors + 1) = 0x335577;
+    SetDIBColorTable( hdcSrc, 0, 2, colors );
+    srcBuffer[0] = 0x55555555;
+    memset(dstBuffer, 0xcc, 4 * sizeof(*dstBuffer));
+    SetTextColor( hdcDst, 0 );
+    SetBkColor( hdcDst, 0 );
+    StretchBlt(hdcDst, 0, 0, 4, 1, hdcSrc, 0, 0, 4, 1, SRCCOPY );
+    expected[0] = expected[2] = 0x00123456;
+    expected[1] = expected[3] = 0x00335577;
+    ok(!memcmp(dstBuffer, expected, 16),
+       "StretchBlt expected { %08X, %08X, %08X, %08X } got { %08X, %08X, %08X, %08X }\n",
+        expected[0], expected[1], expected[2], expected[3],
+        dstBuffer[0], dstBuffer[1], dstBuffer[2], dstBuffer[3] );
+
+    SelectObject(hdcSrc, oldSrc);
+    DeleteObject(bmpSrc);
+
+    bmpSrc = CreateBitmap( 16, 16, 1, 1, 0 );
+    oldSrc = SelectObject(hdcSrc, bmpSrc);
+    SetPixel( hdcSrc, 0, 0, 0 );
+    SetPixel( hdcSrc, 1, 0, 0xffffff );
+    SetPixel( hdcSrc, 2, 0, 0xffffff );
+    SetPixel( hdcSrc, 3, 0, 0 );
+    memset(dstBuffer, 0xcc, 4 * sizeof(*dstBuffer));
+    SetTextColor(hdcDst, RGB(0x22, 0x44, 0x66));
+    SetBkColor(hdcDst, RGB(0x65, 0x43, 0x21));
+    StretchBlt(hdcDst, 0, 0, 4, 1, hdcSrc, 0, 0, 4, 1, SRCCOPY );
+    expected[0] = expected[3] = 0x00224466;
+    expected[1] = expected[2] = 0x00654321;
+    ok(!memcmp(dstBuffer, expected, 16),
+       "StretchBlt expected { %08X, %08X, %08X, %08X } got { %08X, %08X, %08X, %08X }\n",
+        expected[0], expected[1], expected[2], expected[3],
+        dstBuffer[0], dstBuffer[1], dstBuffer[2], dstBuffer[3] );
+
+    SelectObject(hdcSrc, oldSrc);
+    DeleteObject(bmpSrc);
+
+    SelectObject(hdcSrc, oldSrc);
+    DeleteObject(bmpSrc);
+    SelectObject(hdcDst, oldDst);
+    DeleteObject(bmpDst);
+
+    memset(&biDst, 0, sizeof(BITMAPINFO));              // Clear our Bitmap to to all zeroes
+
+    biDst.bmiHeader.biSize = sizeof(BITMAPINFOHEADER);  // Set our Size to the header size
+    biDst.bmiHeader.biWidth = 2;                        // Set our Width to 2 (left-to-right)
+    biDst.bmiHeader.biHeight = -2;                      // Set our Height to -2 that's negative for top-down
+    biDst.bmiHeader.biPlanes = 1;                       // Set out planes to 1 (1 required by Windows)
+    biDst.bmiHeader.biBitCount = 32;                    // Set out BitCount to 32 (Full Color)
+    biDst.bmiHeader.biCompression = BI_RGB;             // Set our Compression to BI_RBG (uncompressed)
```

Comment:
`BI_RGB` is zero.

Reply:
Yes, BI_RGB is zero here. Are you suggesting that this be added in the comment or something else? Thanks.

Reply:
The structure is zero-filled before population. Why do we zero twice?

Reply:
It makes the code self-documented. (So technically these comments are not needed as well.)
Zero-initialize everything so we are sure that even if the structure contains more fields, all will be initialized.
Then only very specific fields (and the important ones) with specific values are initialized.

Reply:
@katahiromz, OK. Now I understand why you noted this. I think that the benefit here is that anyone looking at this code can easily see what the zero means, which is that we are using BI_RGB compression which is "No compression" as indicated from the comment. It also helps to show that we have touched ALL of the fields so that we have covered everything that we intended to initialize. Thanks.